### PR TITLE
FIX skip status display if remaining time is negative

### LINKED ISSUE #327: Number of task done exceeds total amount of tasks
I run Parallel job - 8 workers, input arguments was list of 2 paths.
Job completed and on the commandline I got:

```
[Parallel(n_jobs=8)]: Done   3 out of   2 | elapsed:  8.9min remaining:  -178.2s
[Parallel(n_jobs=8)]: Done   2 out of   2 | elapsed:  8.9min finished
```

3 out of 2 does not make sense

I am using python 2.7.11 in the virtual evironmnet on Mac - El Captain

### PR DESCRIPTION
**Pull request by aabadie**
This should fix #327 and only consists to skip erroneous reporting if remaining time is negative.

### CODE REVIEW

**Comment by ogrisel** [comment]
I think the real problem is that `self.n_dispatched_tasks - index - 1` should never be zero. So there is an issue with the way we manage the `index` variable.

**Comment by aabadie** [comment]
@ogrisel, I pushed an update of the code that try to better fix the issue. The problem comes from the consumption of the `_original_iterator` which is difficult to try.
I have a better output now using a check on number of remaining jobs (`self._jobs`) and `_iterating` attribute of the Parallel object.

**Comment by aabadie** [comment]
@ogrisel, I pushed an update on this : the `index` variable now contains `self.n_completed_tasks`.

File: joblib/parallel.py [hunk 1]
```diff
@@ -622,13 +622,12 @@ def print_progress(self):
...
+        if len(self._jobs) > 0 or getattr(self, "_iterating", False):
             if _verbosity_filter(self.n_dispatched_batches, self.verbose):
                 return
             self._print('Done %3i tasks      | elapsed: %s',
                         (self.n_completed_tasks,
-                         short_format_time(elapsed_time),
-                        ))
+                         short_format_time(elapsed_time), ))
         else:
             index = self.n_dispatched_batches
```
Comment (ogrisel) [inline joblib/parallel.py]: this should be `self.n_completeted_tasks` instead.

File: joblib/parallel.py [hunk 2]
```diff
@@ -622,13 +622,12 @@ def print_progress(self):
 
         # This is heuristic code to print only 'verbose' times a messages
         # The challenge is that we may not know the queue length
-        if self._original_iterator:
```
Comment (ogrisel) [inline joblib/parallel.py]:
```python
if self._original_iterator is not None:
```

is the good check to test whether all tasks have been dispatched or not.

File: joblib/parallel.py [hunk 3]
```diff
@@ -637,20 +636,21 @@ def print_progress(self):
...
                 if (is_last_item or cursor % frequency):
                     return
-            remaining_time = (elapsed_time / (index + 1) *
-                              (self.n_dispatched_tasks - index - 1.))
+            remaining_time = (elapsed_time / index) * \
+                             (self.n_dispatched_tasks - index * 1.0)
+            # only display status if remaining time is greater or equal to 0
             self._print('Done %3i out of %3i | elapsed: %s remaining: %s',
-                        (index + 1,
+                        (index,
```
Comment (ogrisel) [inline joblib/parallel.py]: indeed this is good.

File: joblib/parallel.py (hunk 2 above)
Comment (aabadie) [inline joblib/parallel.py]: seems that you commented on the outdated diff

**Comment by ogrisel** [comment]
Looks great, thanks.

**Pull request merged**